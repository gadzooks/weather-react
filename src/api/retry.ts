// https://www.chrisarmstrong.dev/posts/retry-timeout-and-cancel-with-fetch

// eslint-disable-next-line no-promise-executor-return
const throwOnTimeout = (timeout: number) => new Promise((_, reject) => setTimeout(() => reject(new Error('Timeout')), timeout));

export const fetchWithTimeout = (url: string, options: any = {}) => {
  const { timeout = 3000, ...remainingOptions } = options;
  console.log(`timeout is ${timeout}`);
  // if the timeout option is specified, race the
  // fetch call
  if (timeout) {
    return Promise.race([fetch(url, remainingOptions), throwOnTimeout(timeout)]);
  }
  return fetch(url, remainingOptions);
};

const fetchWithRetries = async (url :string, options :any, retryCount = 0) : Promise<any> => {
  // split out the maxRetries option from the remaining
  // options (with a default of 3 retries)
  const { maxRetries = 3, ...remainingOptions } = options;
  try {
    console.log(`retry ${retryCount}`);
    return await fetchWithTimeout(url, remainingOptions);
  } catch (error) {
    // if the retryCount has not been exceeded, call again
    if (retryCount < maxRetries) {
      console.log(`retry ${retryCount}`);
      return fetchWithRetries(url, options, retryCount + 1);
    }
    // max retries exceeded
    throw error;
  }
};

export default fetchWithRetries;
