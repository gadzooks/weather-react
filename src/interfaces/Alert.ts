interface Alert {
  id: string;
  event: string;
  headline: string;
  description: string;
  ends: string;
  endsEpoch: number;
  link: string;
}

export default Alert;
