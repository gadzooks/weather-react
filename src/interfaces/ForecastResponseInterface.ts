import { LocationInterface } from './LocationInterface';
import { RegionInterface } from './RegionInterface';
import { DailyForecastInterface } from './DailyForecastInterface';

export interface RegionById {
    [key: string]: RegionInterface;
}

export interface RegionsById {
  byId: RegionById;
  allIds: string[];
}

export interface LocationById {
  [key: string]: LocationInterface;
}

export interface ForecastById {
  [key: string]: DailyForecastInterface[];
}

export interface ForecastsById {
  byId: ForecastById;
}

export interface LocationsById {
  byId: LocationById;
  allIds: string[];
}

export interface ForecastResponse {
  dates: string[];
  regions: RegionsById;
  locations: LocationsById;
  forecasts: ForecastsById;
}
export interface ForeacastDates {
  dates: string[];
  parsedDates: Date[];
  weekends: boolean[];
}

export const DefaultForecastDates: ForeacastDates = {
  dates: [],
  parsedDates: [],
  weekends: [],
};

export interface ForecastResponseStatus {
  isLoaded?: boolean;
  error: Error | null;
  forecast: ForecastResponse | null;
  forecastDates: ForeacastDates;
}

export const DefaultForecastResponseStatus: ForecastResponseStatus = {
  isLoaded: false,
  error: null,
  forecast: null,
  forecastDates: DefaultForecastDates,
};

// dates: [
//     "2021-04-17T07:00:00.000+00:00",
//     "2021-04-18T07:00:00.000+00:00",
//     "2021-04-19T07:00:00.000+00:00",
//     "2021-04-20T07:00:00.000+00:00",
// ],
// regions: {
//     byId: {
//     central_cascades: {
//         id: "central_cascades",
//         description: "Central Cascades",
//         regionId: "central_cascades",
//         locations: ["renton", "yakima"],
//     },
//     western_wa: {
//         id: "western_wa",
//         description: "Western WA",
//         regionId: "western_wa",
//         locations: ["seattle", "bellevue"],
//     },
//     },
//     allIds: ["central_cascades", "western_wa"],
// },
// locations: {
//     byId: {
//     renton: {
//         id: "renton",
//         description: "Renton",
//         regionId: "central_cascades",
//     },
//     yakima: {
//         id: "yakima",
//         description: "Yakima",
//         regionId: "central_cascades",
//     },
// forecasts: {
//     byId: {
//       renton: [
//         {
//           time: "2021-04-17T07:00:00.000+00:00",
//           summary: "Clear conditions throughout the day.",
//           icon: "day-hail",
//           precipProbability: 0.0,
//           temperature: 60.9,
//           apparentTemperature: 60.9,
//           dewPoint: 40.1,
//           visibility: 120,
//           cloudCover: 1,
//           temperatureHigh: 74.9,
//           temperatureLow: 47.3,
//           sunsetTime: 1618714951,
//           sunriseTime: 1618665359,
//           precipAmount: 0.0,
//         },
