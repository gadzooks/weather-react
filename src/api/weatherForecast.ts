import { DefaultForecastResponseStatus, ForecastResponseStatus } from '../interfaces/ForecastResponseInterface';

export interface GetForecastProps {
  dataSource: string,
  setAppState: any,
}

const dataSource = import.meta.env.NODE_ENV === 'production' ? 'real' : 'mock';
const WEATHER_API = import.meta.env.VITE_WEATHER_API;
const WEATHER_JWT_TOKEN = import.meta.env.VITE_WEATHER_JWT_TOKEN;
const url = `${WEATHER_API}/forecasts/${dataSource}`;

export async function getForecastAsync(): Promise<any> {
  const results = await fetch(`${url}`, {
    mode: 'cors',
    headers: new Headers({
      Authorization: `Bearer ${WEATHER_JWT_TOKEN}`,
    }),
  });

  const forecast = results.json();
  return forecast;
}

const getForecast = async () : Promise<ForecastResponseStatus> => {
  // eslint-disable-next-line no-promise-executor-return
  // await new Promise((r) => setTimeout(r, 300000));
  await fetch(`${url}`, {
    mode: 'cors',
    headers: new Headers({
      Authorization: `Bearer ${WEATHER_JWT_TOKEN}`,
    }),
  })
    .then((res) => {
      if (!res.ok) {
        throw new Error('bad response from server ');
      }
      return res.json();
    })
    .then(
      (result) => {
        const newAppState: ForecastResponseStatus = {
          isLoaded: true,
          forecast: result.data,
          error: null,
        };
        return newAppState;
      },
      (error) => {
        throw new Error(error);
      },
    ).catch((err) => {
      console.log(`catching error : ${err}`);
      const errorAppState: ForecastResponseStatus = {
        isLoaded: false,
        error: err,
        forecast: null,
      };
      return errorAppState;
    });

  return DefaultForecastResponseStatus;
};

export default getForecast;
