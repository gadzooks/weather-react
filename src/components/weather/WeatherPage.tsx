import { format } from "fecha";
import React, { useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import { ForecastResponse } from "../../interfaces/ForecastResponseInterface";
import SearchableTableHook from "./forecast_summary/SearchableTableHook";

export function isWeekend(dates: (Date|null)[]) :boolean[] {
    const weekends: boolean[] = [];
    if(!dates) {
        return weekends;
    }

    dates.map((d) => {
        if (d) {
            const dayOfWeek = format(d, 'ddd').toUpperCase();
            weekends.push(dayOfWeek === 'SUN' || dayOfWeek === 'SAT')
        } else {
            weekends.push(false);
        }
    })

    return weekends; 
}

interface AppState {
  isLoaded: boolean,
  error: Error|null,
  forecast: ForecastResponse|null,
}

const WeatherPage = (props: AppState) => {

  const [appState, setAppState] = useState(props);
  const params = useParams();
  const dataSource = params.dataSource || 'useLocal';

  useEffect(() => {
    const WEATHER_API = process.env.REACT_APP_WEATHER_API;

    const url = `${WEATHER_API}/forecasts/${dataSource}`;
    console.log(`getting weather from ${url}`);
    fetch(`${url}`, { mode: 'cors' })
      .then(res => res.json())
      .then(
        (result) => {
          setAppState({
            isLoaded: true,
            forecast: result.data,
            error: null,
          });
        },
        // Note: it's important to handle errors here
        // instead of a catch() block so that we don't swallow
        // exceptions from actual bugs in components.
        (error) => {
          setAppState({
            isLoaded: true,
            error: error,
            forecast: null,
          });
        }
      )
  }, [])

  return(
    <>
    { appState.error && <div>Error: {appState.error.message}</div> }
    { !appState.isLoaded && <div>Loading...</div> }
    { appState.forecast && <Page {...appState.forecast} /> }
    </>
  );
}

export default WeatherPage;

export function Page(props: ForecastResponse) {
  return (
    <>
      <div id='top' />
      <SearchableTableHook {...props} />
    </>
  )

}