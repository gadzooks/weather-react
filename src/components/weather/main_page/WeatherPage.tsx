import React, { useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import { ForecastResponse } from "../../../interfaces/ForecastResponseInterface";
import SearchableTableHook from "../forecast_summary/SearchableTableHook";

interface AppState {
  isLoaded: boolean,
  error: Error|null,
  forecast: ForecastResponse|null,
}

const WeatherPage = (props: AppState) => {

  const [appState, setAppState] = useState(props);
  const params = useParams();
  const dataSource = params.dataSource || 'useLocal';

  useEffect(() => {
    const WEATHER_API = process.env.REACT_APP_WEATHER_API;

    const url = `${WEATHER_API}/forecasts/${dataSource}`;
    console.log(`getting weather from ${url}`);
    fetch(`${url}`, { mode: 'cors' })
      .then(res => res.json())
      .then(
        (result) => {
          setAppState({
            isLoaded: true,
            forecast: result.data,
            error: null,
          });
        },
        // Note: it's important to handle errors here
        // instead of a catch() block so that we don't swallow
        // exceptions from actual bugs in components.
        (error) => {
          setAppState({
            isLoaded: true,
            error: error,
            forecast: null,
          });
        }
      )
  }, [])

  return(
    <>
    { appState.error && <div>Error: {appState.error.message}</div> }
    { !appState.isLoaded && <div>Loading...</div> }
    { appState.forecast && <Page {...appState.forecast} /> }
    </>
  );
}

export default WeatherPage;

export function Page(props: ForecastResponse) {
  return (
    <>
      <div id='top' />
      <SearchableTableHook {...props} />
    </>
  )

}