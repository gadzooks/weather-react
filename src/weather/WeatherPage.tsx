import { parse, format } from "fecha";
import React from "react";
import { ForecastResponse } from "../interfaces/ForecastResponseInterface";
import SearchableTableHook from "./forecast_summary/SearchableTableHook";
import LocationDetails from "./location_details/LocationDetails";

function isWeekend(dates: (Date|null)[]) :boolean[] {
    const weekends: boolean[] = [];
    if(!dates) {
        return weekends;
    }

    dates.map((d) => {
        if (d) {
            const dayOfWeek = format(d, 'ddd').toUpperCase();
            weekends.push(dayOfWeek === 'SUN' || dayOfWeek === 'SAT')
        } else {
            weekends.push(false);
        }
    })

    return weekends; 
}

interface AppState {
  isLoaded: boolean,
  error: Error|null,
  forecast: ForecastResponse|null,
}

class WeatherPage extends React.Component<AppState, AppState> {
  constructor(props: AppState) {
    super(props);
    this.state = {
      isLoaded: false,
      error: null,
      forecast: null,
    }
  }

  componentDidMount() {
    // const result = mockWeatherForecastNormalized();
    // this.setState({
    // })
    const WEATHER_API = process.env.REACT_APP_WEATHER_API;

    //TODO: get params from url and decide url vs mock
    fetch(`${WEATHER_API}/forecasts/mock`, {mode:'cors'})
      .then(res => res.json())
      .then(
        (result) => {
          this.setState({
            isLoaded: true,
            forecast: result.data,
            error: null,
          });
        },
        // Note: it's important to handle errors here
        // instead of a catch() block so that we don't swallow
        // exceptions from actual bugs in components.
        (error) => {
          this.setState({
            isLoaded: true,
            error
          });
        }
      )
  }

  render() {
    const { error, isLoaded, forecast } = this.state;
    if (error) {
      return <div>Error: {error.message}</div>;
    } else if (!isLoaded) {
      return <div>Loading...</div>;
    } else if (forecast != null) {
      return <Page {...forecast} />
    }
  }
}

function Page(props: ForecastResponse) {
      const parsedDates = props.dates.map((d) => parse(d, 'YYYY-MM-DD'));
      const weekends = isWeekend(parsedDates);
      const args = {
        ...props,
        weekends,
        parsedDates,
      };

      return (
        <>
          <SearchableTableHook isWeekend={weekends} {...args}/>
          <LocationDetails forecastsByName={props.forecasts.byId} isWeekend={weekends} locations={props.locations.allIds} dates={parsedDates} />
        </>
      )

}

export default WeatherPage;