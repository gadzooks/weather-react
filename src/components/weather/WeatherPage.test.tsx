import React from 'react';
import renderer from 'react-test-renderer';
import { mockWeatherForecastNormalized } from '../../api/mockData';
import { ForecastResponse } from '../../interfaces/ForecastResponseInterface';
import { Page } from './WeatherPage';

it('renders a snapshot', () => {
    const mockData = mockWeatherForecastNormalized().data as unknown as ForecastResponse;
    // const parsedDates = mockData.dates.map((d) => parse(d, 'YYYY-MM-DD'));
    // const weekends = isWeekend(parsedDates);
    // const seattle = mockData.regions.locations[0];
    const tree = renderer.create(
        <Page {...mockData} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});