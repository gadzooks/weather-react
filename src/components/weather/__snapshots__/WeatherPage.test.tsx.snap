// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
Array [
  <div
    id="top"
  />,
  <div
    className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="outlined-basic"
      id="outlined-basic-label"
    >
      Search Locations
    </label>
    <div
      className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-describedby="outlined-basic-helper-text"
        aria-invalid={false}
        autoFocus={true}
        className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        defaultValue=""
        disabled={false}
        id="outlined-basic"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-1ftyaf0"
        >
          <span>
            Search Locations
          </span>
        </legend>
      </fieldset>
    </div>
    <p
      className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
      id="outlined-basic-helper-text"
    >
      No matching locations
    </p>
  </div>,
  <table
    className="MuiTable-root table table-sm weather-forecast-summary css-rqglhn-MuiTable-root"
    role={null}
  >
    <thead
      className="MuiTableHead-root table-heading css-15wwp11-MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          Weather Alerts
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          Location
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          SAT APR 17
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          SUN APR 18
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          MON APR 19
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          TUE APR 20
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          WED APR 21
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          THU APR 22
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          FRI APR 23
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          SAT APR 24
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          SUN APR 25
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          MON APR 26
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          TUE APR 27
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          WED APR 28
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          THU APR 29
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          FRI APR 30
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          SAT MAY 01
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root region-details css-1q1u3t4-MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
          colSpan={17}
        >
          Cities
        </td>
      </tr>
      <tr
        className="MuiTableRow-root weather-cell css-1q1u3t4-MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell css-1ex1afd-MuiTableCell-root"
        >
          N/A
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell css-1ex1afd-MuiTableCell-root"
        >
          <a
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
            href="#seattle"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            SEATTLE CITY
          </a>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center  weekend  css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-day-sunny day-sunny "
            title="Clear conditions throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center  weekend  css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-day-sunny day-sunny "
            title="Clear conditions throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-day-sunny day-sunny "
            title="Clear conditions throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-day-cloudy day-cloudy "
            title="Partly cloudy throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-day-cloudy day-cloudy "
            title="Partly cloudy throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-rain rain "
            title="Partly cloudy throughout the day with a chance of rain."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-rain rain "
            title="Partly cloudy throughout the day with morning rain."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center  weekend  css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-rain rain "
            title="Cloudy skies throughout the day with a chance of rain."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center  weekend  css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-rain rain "
            title="Cloudy skies throughout the day with a chance of rain throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-rain rain "
            title="Partly cloudy throughout the day with rain in the morning and afternoon."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-cloudy cloudy sunshine-50"
            title="Partly cloudy throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-cloudy cloudy sunshine-50"
            title="Cloudy skies throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-cloudy cloudy sunshine-50"
            title="Partly cloudy throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-cloudy cloudy sunshine-50"
            title="Cloudy skies throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center  weekend  css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-cloudy cloudy sunshine-10"
            title="Partly cloudy throughout the day."
          />
        </td>
      </tr>
      <tr
        className="MuiTableRow-root weather-cell css-1q1u3t4-MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell css-1ex1afd-MuiTableCell-root"
        >
          N/A
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell css-1ex1afd-MuiTableCell-root"
        >
          <a
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
            href="#bonney_lake"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            BONNEY LAKE
          </a>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center  weekend  css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-day-sunny day-sunny "
            title="Clear conditions throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center  weekend  css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-day-sunny day-sunny "
            title="Clear conditions throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-day-sunny day-sunny "
            title="Clear conditions throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-rain rain "
            title="Partly cloudy throughout the day with rain in the morning and afternoon."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-day-cloudy day-cloudy "
            title="Partly cloudy throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-rain rain "
            title="Partly cloudy throughout the day with a chance of rain."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-rain rain "
            title="Partly cloudy throughout the day with rain."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center  weekend  css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-rain rain "
            title="Cloudy skies throughout the day with rain."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center  weekend  css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-rain rain "
            title="Cloudy skies throughout the day with a chance of rain."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-rain rain "
            title="Cloudy skies throughout the day with a chance of rain."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-rain rain "
            title="Partly cloudy throughout the day with a chance of rain."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-rain rain "
            title="Cloudy skies throughout the day with rain in the morning and afternoon."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-cloudy cloudy sunshine-50"
            title="Partly cloudy throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-cloudy cloudy sunshine-50"
            title="Cloudy skies throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center  weekend  css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-cloudy cloudy sunshine-10"
            title="Clearing in the afternoon."
          />
        </td>
      </tr>
    </tbody>
  </table>,
]
`;
